File: Java/HashMap/DiffII.java
public class Solution {
	public int diffPossible(final List<Integer> a, int b) {
	    HashSet<Integer> map = new HashSet<>();
	    for(int i = 0; i < a.size(); i++){
	        int val = a.get(i);
	        if(map.contains(val-b) || map.contains(val+b))return 1;
	        map.add(val);
	    }
	    return 0;
	}
}

File: JS/HashMap/DiffII.js
module.exports = {
	//param A : array of integers
	//param B : integer
	//return an integer
	diffPossible : function(A, B){
        var map = {};
        for(var i = 0; i < A.length; i++){
            var val = A[i];
            if(map.hasOwnProperty(val - B) || map.hasOwnProperty(val + B))
                return 1;
            map[val] = true;
        }
        return 0;
	}
};
